### DISCUSSION TITLE
Can not run on arm RK3568

### DISCUSSION THREAD

**Discussion by zrz4066** [slint-ui/slint#3646] [Q&A]
The demo run ok on wsl and windows.But can't run on rk3568
slint1.2.2

libGL error: failed to create dri screen
libGL error: failed to load driver: rockchip
libGL error: failed to create dri screen
libGL error: failed to load driver: rockchip
thread 'main' panicked at 'Failed to set icons: XError { description: "GLXBadContextTag", error_code: 162, request_code: 152, minor_code: 16 }', /home/rpdzkj/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.28.7/src/platform_impl/linux/x11/window.rs:1008:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

![Screenshot from 2023-10-11 16-02-50](https://github.com/slint-ui/slint/assets/16775484/b1471618-7605-42c3-95a7-34e3a43224ab)

Simple demo run ok

![Screenshot from 2023-10-11 16-53-10](https://github.com/slint-ui/slint/assets/16775484/9bba9a81-09b3-43bc-8183-567f0e3e735e)

**Comment by hunger** [comment]
So the RK3568 is an embedded board running ubuntu Linux?

Looking at the screenshot: Looks to me like some graphics-related driver is not available. What kind of environment do you plan to run on? Wayland with DRI?

I'd start with trying some other application using 3D acceleration. If that does not come up either: Check the packages in ubuntu for anything DRI related and try installing those. Sorry, my ubuntu-foo is a bit limited, especially on ARM hardware.

https://forum.radxa.com/t/ubuntu-20-02-mali-drivers/12373/6 might help, but I am not sure ;-)

**Reply by zrz4066** [reply]
Here is the screen on Sep 15,but after running cargo clean,rebuild the demo,it doesnt't work well(slint ="1.2.2" as well)

thread 'main' panicked at 'Failed to set icons: XError { description: "GLXBadContextTag", error_code: 156, request_code: 146, minor_code: 16 }', /home/rpdzkj/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.28.7/src/platform_impl/linux/x11/window.rs:1008:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


![图片](https://github.com/slint-ui/slint/assets/16775484/ea172077-615c-4b53-a7b7-46328540a9ea)
The folllowing is  Cargo.toml same as Sep 15  
[package]
name = "myguidemo"
version = "0.1.0"
authors = ["ice"]
edition = "2021"
build = "build.rs"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html


[dependencies]
#slint ="1.2.2"
slint = { git = "https://github.com/slint-ui/slint", version = "*"}
#slint = { version = "*", features = ["renderer-winit-software"] }
[build-dependencies]
#slint-build = "1.2.2"
slint-build = { git = "https://github.com/slint-ui/slint", version = "*"}

**Reply by ogoffart** [reply]
the crash happens in [winit](https://github.com/rust-windowing/winit). Not sure we can do much about it in Slint. 
Since the error message hints about icons, are you by any chance setting a window icon? (by default, there is none but that shouldn't be a problem)
But the fact is that seems to be related to the GL driver. 
You can also try to use  `SLINT_BACKEND=winit-software` to use the software renderer instead of OpenGL

**Reply by tronical** [reply]
Another idea: In the configuration of your X server, could you try adding a section to disable GLX? Like this:

```
Section "Extensions"
    Option "GLX" "Disable"
EndSection
```

You can put that into a file (say `disable-glx.conf`) and drop it in `/etc/X11/xorg.conf.d/`. After that you need to restart your X server.
